File: Algorithms/C++/00217-Contains Duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        std::unordered_set<int> set;

        for(int i{0}; i < nums.size(); i++){
            if(set.contains(nums[i])) return true;
            set.insert(nums[i]);
        }

        return false;
    }
};

/*
    - use a set
    - run through nums
    - add to set if number is not in set
    - return number if number is already in set
*/

File: Algorithms/Python3/00217-Contains Duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        mySet = set()
        for num in nums:
            if num in mySet:
                return True
            mySet.add(num)
        
        return False

'''
use a hash set
    iterate through nums
    check set if num exists
        if doesn't exist, add to set
        if exists, return true
    
    return false

'''